### PR TITLE
[5.2] Make type hints match doc blocks

### PR DESCRIPTION
**Pull request by greydnls**
This makes the doc blocks and type hints match.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Thank you, but these changes are either wrong, or breaking. I'll leave some online comments soon.

File: src/Illuminate/Queue/RedisQueue.php [hunk 1]
```diff
@@ -44,7 +44,6 @@ class RedisQueue extends Queue implements QueueContract
      * @param  \Illuminate\Redis\Database  $redis
      * @param  string  $default
      * @param  string  $connection
-     * @return void
```
Comment (GrahamCampbell) [inline src/Illuminate/Queue/RedisQueue.php]: Sorry, not what we want.

File: src/Illuminate/Queue/RedisQueue.php [hunk 2]
```diff
@@ -120,8 +119,8 @@ public function release($queue, $payload, $delay, $attempts)
     /**
      * Pop the next job off of the queue.
      *
-     * @param  string  $queue
-     * @return \Illuminate\Contracts\Queue\Job|null
+     * @param  string|null  $queue
```
Comment (GrahamCampbell) [inline src/Illuminate/Queue/RedisQueue.php]: This change is good, but I think is not the only place where this need fixing. A separate PR to fix those would be greatly appreciated. :)

File: src/Illuminate/Queue/RedisQueue.php [hunk 3]
```diff
@@ -120,8 +119,8 @@ public function release($queue, $payload, $delay, $attempts)
...
+     * @return RedisJob|null
```
Comment (GrahamCampbell) [inline src/Illuminate/Queue/RedisQueue.php]: This was fine as it was. We want to exactly match the contract definition.

File: src/Illuminate/Queue/RedisQueue.php [hunk 4]
```diff
@@ -205,7 +204,7 @@ public function migrateExpiredJobs($from, $to)
      * @param  int  $time
      * @return array
      */
-    protected function getExpiredJobs($transaction, $from, $time)
+    protected function getExpiredJobs(\Predis\Transaction\MultiExec $transaction, $from, $time)
```
Comment (GrahamCampbell) [inline src/Illuminate/Queue/RedisQueue.php]: Adding a typehint to a public or protected method is breaking. Please send this and the related changes to the 5.3 branch, and also can you impprt this at the top please?

File: tests/Queue/QueueRedisQueueTest.php [hunk 5]
```diff
@@ -86,7 +86,7 @@ public function testMigrateExpiredJobs()
     {
         $queue = $this->getMock('Illuminate\Queue\RedisQueue', ['getTime'], [$redis = m::mock('Illuminate\Redis\Database'), 'default']);
         $queue->expects($this->once())->method('getTime')->will($this->returnValue(1));
-        $transaction = m::mock('StdClass');
+        $transaction = m::mock(\Predis\Transaction\MultiExec::class);
```
Comment (GrahamCampbell) [inline tests/Queue/QueueRedisQueueTest.php]: See my previous note. :)

File: src/Illuminate/Queue/RedisQueue.php (hunk 3 above)
Comment (greydnls) [inline src/Illuminate/Queue/RedisQueue.php]: It's factually inaccurate.

**Pull request closed without merging**